fastai/callback/tensorboard.py

```diff
@@ -42,8 +43,98 @@ def after_epoch(self):
             x,y,its,outs = self.dls.valid.show_results(b, out, show=False, max_n=self.n_preds)
             tensorboard_log(x, y, its, outs, self.writer, self.train_iter)
 
+    def before_validate(self):
+        if self.projector:
```

Comment:
Let's reduce nesting by making `if not self.projector: return` the first line of each of these methods